File: python/strings/Reverse_Prefix_of_Word_2000.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        i=0
        for char in word:
            if char==ch:
                prefix=word[:i+1]
                suffix=word[i+1:]
                rev_prefix=prefix[::-1]
                return rev_prefix+suffix
            else:
                i+=1
        return word
            

File: java/strings/Reverse_Prefix_of_Word_2000.java
package java.strings;

public class Reverse_Prefix_of_Word_2000 {
        public String reversePrefix(String word, char ch) {

       int index=word.indexOf(ch);
       if(index==-1){
        return word;
       }
       String prefix=word.substring(0,index+1);
       String suffix=word.substring(index+1,word.length());
       StringBuilder reversed=new StringBuilder(prefix);
       reversed.reverse();

        return reversed.toString()+suffix;
    }
}
